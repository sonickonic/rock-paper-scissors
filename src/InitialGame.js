import React from "react";
import styled from "styled-components";

import shapes from "./data";
import Background from "./images/bg-triangle.svg";

const Container = styled.div`
  background-image: url(${Background});
  background-repeat: no-repeat;
  background-size: 190px;
  background-position: center;
  max-width: 315px;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
`;

const Chip = styled.div`
  position: relative;
  background: linear-gradient(
    ${(props) => props.lightColor},
    ${(props) => props.darkColor}
  );
  width: 13.4rem;
  height: 13.4rem;
  border-radius: 100%;
  box-shadow: inset 0px -6px 0px rgba(0, 0, 0, 0.25);

  :last-child {
    margin: 1.5rem auto 0;
  }

  ::before {
    content: "";
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    width: 10rem;
    height: 10rem;
    border-radius: 100%;
    background-color: var(--white);
    box-shadow: inset 0px 4px 0px rgba(0, 0, 0, 0.25);
  }
`;

const Shape = styled.img`
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
`;

function InitialGame() {
  return (
    <Container>
      {shapes.map((shape) => (
        <Chip
          key={shape.name}
          lightColor={shape.lightColor}
          darkColor={shape.darkColor}
        >
          <Shape src={shape.imageSrc} alt={shape.name} />
        </Chip>
      ))}
    </Container>
  );
}

export default InitialGame;
