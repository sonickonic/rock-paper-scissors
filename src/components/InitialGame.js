import React from "react";
import shapes from "../data";
import Hand from "./Hand";

const InitialGame = ({ currentGame, handleSelection }) => {
  return (
    <currentGame.container currentGame={currentGame}>
      {shapes
        .filter((shape) => currentGame.shapes.includes(shape.name))
        .map((shape) => (
          <Hand
            shape={shape}
            key={shape.name}
            onClick={() => {
              handleSelection(shape);
            }}
            size={currentGame.size}
          />
        ))}
    </currentGame.container>
  );
};

export default InitialGame;
