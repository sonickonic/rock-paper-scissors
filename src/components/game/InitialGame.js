import React from "react";
import shapes from "../../data";
import Hand from "./Hand";
import { connect } from "react-redux";

const InitialGame = ({ currentGame, handleSelection }) => {
  return (
    <currentGame.container>
      {shapes
        .filter((shape) => currentGame.shapes.includes(shape.name))
        .map((shape) => (
          <Hand
            shape={shape}
            key={shape.name}
            onClick={() => {
              handleSelection(shape);
            }}
            size={currentGame.size}
          />
        ))}
    </currentGame.container>
  );
};

const mapStateToProps = (state) => ({
  currentGame: state.game.currentGame,
});

export default connect(mapStateToProps)(InitialGame);
