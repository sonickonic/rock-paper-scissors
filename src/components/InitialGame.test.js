import React from "react";
import { render } from "@testing-library/react";
import InitialGame from "./InitialGame";
import gameType from "../gameTypes";

describe("InitialGame", () => {
  describe("classic game", () => {
    const classic = gameType[0];

    it("renders correctly", () => {
      const { asFragment } = render(<InitialGame currentGame={classic} />);

      expect(asFragment()).toMatchSnapshot();
    });

    it("shows three hands", () => {
      const { container } = render(<InitialGame currentGame={classic} />);

      expect(container.querySelectorAll("img")).toHaveLength(3);
    });
  });

  describe("expansion game", () => {
    const expansion = gameType[1];

    it("renders correctly", () => {
      const { asFragment } = render(<InitialGame currentGame={expansion} />);

      expect(asFragment()).toMatchSnapshot();
    });

    it("shows five hands", () => {
      const { container } = render(<InitialGame currentGame={expansion} />);

      expect(container.querySelectorAll("img")).toHaveLength(5);
    });
  });
});
