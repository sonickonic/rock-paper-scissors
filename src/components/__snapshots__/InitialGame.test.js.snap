// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InitialGame classic game renders correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-AxiKw bRGait"
  >
    <div
      class="sc-AxjAm bVDfZd"
    >
      <img
        alt="paper"
        class="sc-AxirZ dtQZHs"
        src="icon-paper.svg"
      />
    </div>
    <div
      class="sc-AxjAm fvgMhd"
    >
      <img
        alt="scissors"
        class="sc-AxirZ dtQZHs"
        src="icon-scissors.svg"
      />
    </div>
    <div
      class="sc-AxjAm ckhsNd"
    >
      <img
        alt="rock"
        class="sc-AxirZ dtQZHs"
        src="icon-rock.svg"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`InitialGame expansion game renders correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-AxhCb hSmDqS"
  >
    <div
      class="sc-AxjAm gQvrnB"
    >
      <img
        alt="paper"
        class="sc-AxirZ eMbVZM"
        src="icon-paper.svg"
      />
    </div>
    <div
      class="sc-AxjAm kEAtnB"
    >
      <img
        alt="scissors"
        class="sc-AxirZ eMbVZM"
        src="icon-scissors.svg"
      />
    </div>
    <div
      class="sc-AxjAm jSYbZd"
    >
      <img
        alt="rock"
        class="sc-AxirZ eMbVZM"
        src="icon-rock.svg"
      />
    </div>
    <div
      class="sc-AxjAm bhcJld"
    >
      <img
        alt="lizard"
        class="sc-AxirZ eMbVZM"
        src="icon-lizard.svg"
      />
    </div>
    <div
      class="sc-AxjAm hvPtrl"
    >
      <img
        alt="spock"
        class="sc-AxirZ eMbVZM"
        src="icon-spock.svg"
      />
    </div>
  </div>
</DocumentFragment>
`;
